src/Checker.ts

```diff
@@ -11,12 +11,16 @@ export const checkerToGuard: <FROM, TO extends FROM, N extends FROM = FROM>(
 );
 
 function getRawGuard<FROM, TO extends FROM>(checker: Checker<FROM>): ReasonGuard<FROM, TO> {
-	return (input, e = [], c = []): input is TO => {
+	return (input, e = [], c = [], context = []): input is TO => {
 		try {
-			c.push(checker(input));
+			c.push(checker(input, context));
 			return true;
 		} catch (err) {
-			e.push(err);
+			if (Array.isArray(err)) {
+				e.push(...err);
+			} else {
+				e.push(err);
+			}
```

Comment:
could be:

```e = e.concat(err)```

I think? It's more concise at least

Reply:
It wont work because we pass in the e array and modify it's content.
Concat returns a new array.